### PR TITLE
DOCSP-33504: distinct returns struct v2

### PR DESCRIPTION
**Pull request by rustagir**
# Pull Request Info

[PR Reviewing Guidelines](https://github.com/mongodb/docs-golang/blob/master/REVIEWING.md)

JIRA - <https://jira.mongodb.org/browse/DOCSP-33504>
Staging:
- [retrieve](https://preview-mongodbrustagir.gatsbyjs.io/golang/DOCSP-33504-distinct-cursor/fundamentals/crud/read-operations/distinct/)
- [usage ex](https://preview-mongodbrustagir.gatsbyjs.io/golang/DOCSP-33504-distinct-cursor/usage-examples/distinct/)
- [whats new](https://preview-mongodbrustagir.gatsbyjs.io/golang/DOCSP-33504-distinct-cursor/whats-new/#what-s-new-in-2.0)

## Self-Review Checklist

- [x] Is this free of any warnings or errors in the RST?
- [x] Did you run a spell-check?
- [x] Did you run a grammar-check?
- [x] Are all the links working?
- [x] Are the [facets and meta keywords](https://wiki.corp.mongodb.com/display/DE/Docs+Taxonomy) accurate?

### CODE REVIEW

**Review by mcmorisi — APPROVED** [review]
LGTM!

File: source/includes/usage-examples/code-snippets/distinct.go [hunk 1]
```diff
@@ -39,16 +39,14 @@ func main() {
...
 	// that match the filter
-	results, err := coll.Distinct(context.TODO(), "title", filter)
+	res := coll.Distinct(context.TODO(), "title", filter)
 
-	// Prints a message if any errors occur during the operation
-	if err != nil {
-		panic(err)
-	}
+	var arr []string
+	res.Decode(&arr)
```
Comment (matthewdale) [inline source/includes/usage-examples/code-snippets/distinct.go:45]: We still need to check the error here. Also, consider using the more common pattern of chaining the `Distinct` call.

For example:
```go
var arr []string
err := coll.Distinct(context.TODO(), "title", filter).Decode(&arr)
if err != nil {
	panic(err)
}
```

File: source/includes/fundamentals/code-snippets/CRUD/distinctValues.go [hunk 2]
```diff
@@ -57,13 +57,12 @@ func main() {
...
-		panic(err)
-	}
+	filter := bson.D{{"enrollment", bson.D{{"$lt", 50}}}}
+	res := coll.Distinct(context.TODO(), "department", filter)
 
-	for _, result := range results {
-		fmt.Println(result)
-	}
+	var arr []string
+	res.Decode(&arr)
```
Comment (matthewdale) [inline source/includes/fundamentals/code-snippets/CRUD/distinctValues.go:64]: We still need to check the error here. Also, consider using the more common pattern of chaining the `Distinct` call.

For example:
```go
var arr []string
err := coll.Distinct(context.TODO(), "department", filter).Decode(&arr)
if err != nil {
	panic(err)
}
```

**Review by matthewdale — CHANGES_REQUESTED** [review]

File: source/includes/usage-examples/code-snippets/distinct.go (hunk 1 above)
Comment (rustagir) [reply source/includes/usage-examples/code-snippets/distinct.go:45]: thanks for clarifying the API, will change!

**Review by matthewdale — APPROVED** [review]
Looks good! 👍

**Pull request merged**